### PR TITLE
Storage: refactor - decouple the Upload method from HTTP concepts

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
A small refactor PR that extracts all HTTP concepts (multipart file, status codes, http content type detection) from the `StorageService.Upload` method and moves them to the HTTP handler found in `http.go`.

Also, `StorageService.Upload` will now error out if a file with the same path already exists, and `UploadRequest.OverwriteExistingFile` is set to false

### CODE REVIEW

File: pkg/services/store/service.go [hunk 1]
```diff
@@ -122,71 +124,78 @@ func isFileTypeValid(filetype string) bool {
...
+	if !strings.HasPrefix(req.Path, RootUpload+"/") {
+		return ErrUnsupportedFolder
+	}
+
+	validFileType := isFileTypeValid(req.MimeType)
+	if !validFileType {
+		return ErrInvalidFileType
+	}
+
+	if (len(req.Contents)) > MAX_UPLOAD_SIZE {
```
Comment (undef1nd) [inline pkg/services/store/service.go:153]: For custom branding we'll need to upload files larger than 1mb. Would it be possible to make it configurable?

File: pkg/services/store/http.go [hunk 2]
```diff
@@ -27,26 +31,95 @@ func ProvideHTTPService(store StorageService) HTTPStorageService {
...
+
+	if errors.Is(err, ErrInvalidFileType) {
+		return 400
+	}
+
+	if errors.Is(err, ErrFileAlreadyExists) {
+		return 400
+	}
+
+	return 500
```
Comment (FZambia) [inline pkg/services/store/http.go:59]: Oh my Go

**Review by FZambia — APPROVED** [review]
This is still under feature toggle, so security aspects is sth we should carefully revisit. A nice refactoring anyway.

File: pkg/services/store/http.go (hunk 2 above)
Comment (ryantxu) [reply pkg/services/store/http.go:59]: I think you can use a switch statement here...  slightly more go idiomatic 🤷‍♂️ 

https://medium.com/@sebdah/go-best-practices-error-handling-2d15e1f0c5ee

**Review by ryantxu — APPROVED** [review]
nice

File: pkg/services/store/http.go (hunk 2 above)
Comment (ArturWierzbicki) [reply pkg/services/store/http.go]: > Oh my Go

Gorror! Refactored to a switch stmnt

File: pkg/services/store/http.go [hunk 3]
```diff
@@ -27,26 +31,95 @@ func ProvideHTTPService(store StorageService) HTTPStorageService {
...
+	// open each file to copy contents
+	file, err := fileHeader.Open()
 	if err != nil {
 		return response.Error(500, "Internal Server Error", err)
 	}
+	err = file.Close()
+	if err != nil {
+		return response.Error(500, "Internal Server Error", err)
+	}
+	data, err := ioutil.ReadAll(file)
```
[the remark this answers is not in the document]
Comment (ArturWierzbicki) [reply pkg/services/store/http.go:92]: Yeah, we already use `maxBytesReader` on 64th line.

**Pull request merged**